lib/RoleCredentialResolver.ts

```diff
@@ -0,0 +1,189 @@
+import { RoleCredentialSubject } from '@energyweb/credential-governance';
+import {
+  CredentialResolver,
+  RoleEIP191JWT,
+  RolePayload,
+  VerifiableCredential,
+} from '@energyweb/vc-verification';
+import { DidStore } from '@ew-did-registry/did-ipfs-store';
+import { RegistrySettings } from '@ew-did-registry/did-resolver-interface';
+import { providers, utils } from 'ethers';
+import { CacheServerClient } from './cacheServerClient';
+import {
+  IpfsCredentialResolver,
+  isEIP191Jwt,
+  isVerifiableCredential,
+  transformClaim,
+  filterOutMaliciousClaims,
+} from '@energyweb/vc-verification';
+import * as jwt from 'jsonwebtoken';
+
+export class RoleCredentialResolver implements CredentialResolver {
```

Comment:
Is it make sense to implement CredentialResolver for resolution only from ssi-hub. Say `SsiHubCredentialResolver`

Reply:
IMO this would extra overhead because if there is no `sis-hub` available then we would have to go for `IpfsCredentialResolver`. So there is extra validation and more classes for same purpose.

Reply:
> So there is extra validation

Do you mean null-check of ssi-hub?

 > more classes for same purpose

CredentialResolver is interface, it is supposed to have different implementations

Reply:
We had `IpfsCredentialResolver` for getting credential from IPFS, but then we wanted to get DID Doc from `sis-hub` so we implemented `RoleCredentialResolver` which now has both ability (either to get DID Doc from blockchain of sis-hub). So if we create `SsiHubCredentialResolver` then also we would have a third class which either gets credential through `SsiHubCredentialResolver` or `IpfsCredentialResolver` keeping a validation that if `sis-hub` is available or not. So I am not actually understanding how this new `SsiHubCredentialResolver` would be efficient?